install/cupy_builder/install_build.py

```diff
@@ -689,9 +693,59 @@ def get_cusparselt_version(formatted=False):
     return _cusparselt_version
 
 
+def conda_get_target_name():
+    out = None
+    if PLATFORM_LINUX:
+        plat = platform.processor()
+        if plat == "aarch64":
+            out = "sbsa-linux"
+        else:
+            out = f"{plat}-linux"
+    else:
+        raise NotImplementedError
+    logging.debug(f"{out=}")
+    return out
+
+
+def conda_update_dirs(include_dirs, library_dirs):
+    # Note: These hacks are needed for the dependency detection stage to function,
+    # because we create a fresh compiler instance that does not honor CFLAGS etc set
+    # in the conda-build environment.
+    include_dirs = list(include_dirs)
+    library_dirs = list(library_dirs)
+
+    if os.environ.get('CONDA_BUILD_CROSS_COMPILATION'):
+        # If we're cross compiling, we need to generate stub files that are
+        # executable in the build environment, not the target environment.
+        # This assumes, however, that the build/host environments see the same
+        # CUDA Toolkit.
```

Comment:
Question: Is it possible for the conda-forge build environment to run [`docker run --rm --privileged multiarch/qemu-user-static --reset -p yes`](https://github.com/multiarch/qemu-user-static) before building CuPy? This should allow the host to run aarch64/ppc64le binaries via QEMU.

This technique was originally used to build Jetson wheels on x86_64. https://github.com/cupy/cupy-release-tools/blob/d588c6f155f1b43ef983f75ffcd556fa6b8f6215/dist.py#L420-L428

Reply:
Thanks, got it. I thought the overhead of running cross-compiled stubs on QEMU could be negligible compared to that of running all the compilation on QEMU. But anyway let's merge as-is, it's working great.

Reply:
Yeah with the 2-core Azure runners it's hard to do any optimization for compilation intensive packages... 😢 Cross compilation is also bad in that it doesn't even allow basic sanity checks (like `import cupy`) and I am not a fan either, but it gets the problem solved for now. (Actually we have a new problem in the feedstock, but I can't get to that this week...) I hope to revisit this when the CF GPU CI is up. At least we'd have a more beefy CPU.